inspirehep/records/api/literature.py

```diff
@@ -44,32 +50,60 @@
 CROSSREF_URL = "https://api.crossref.org/works/<ID>"
 
 
-class LiteratureRecord(InspireRecord, CitationMixin):
+class LiteratureRecord(FilesMixin, CitationMixin, InspireRecord):
     """Literature Record."""
 
     es_serializer = LiteratureESEnhancementV1
     pid_type = "lit"
     pidstore_handler = PidStoreLiterature
 
+    @property
+    def earliest_date(self):
```

Comment:
`CitationMixin` is the only code which is using `earliest_date` method, it won't work without it. Is it wise to put this method directly in `LiteratureRecord` then and not in `CitationMixin`?

Reply:
https://github.com/inspirehep/inspirehep/blob/78f55645b67e2fe39ff37611671c263cd9a79e54/inspirehep/records/marshmallow/literature/base.py#L220

Reply:
1. we're filtering out all other collections https://github.com/inspirehep/inspirehep/blob/e37dde111f8396491d966789e30ce073870841d5/inspirehep/records/api/base.py#L1037
2. ``Data`` doesn't call ``_update_refs_in_citation_table``